### DISCUSSION TITLE
prevent merge for contributers in a private repository

### DISCUSSION THREAD

**Discussion by benyamin-nayudi** [community/community#30875] [Other Feature Feedback, Questions, & Ideas]
Hi, I have an issue related to private repositories. i have a company account and we have some private repositories that just a specific teams have access to them. we wanted to start a private project on a repo but the members either should get a Read or Triage access or Write access. i want my team members to see the repo, to be able to make a branch and send a PR without allowing them to have the ability to merge their PRs. if I give them the Read or Triage access they can't make branches and if I give them Write access they will be able to merge their codes. is there any configurations or solutions to this problem?

the solution untill now is to make the repo public but i just want a specific members have the ability to see the repo.

**Comment by amir-mostafa-hs** [comment]
Hello Mr Nayudi. 
###### " [Repository roles for an organization](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization)  "
I think this link can help you to solve this problem.
